composer/loggers/remote_uploader_downloader.py

```diff
@@ -463,9 +466,19 @@ def wait_for_workers(self):
                 if len(self._logged_objects) == 0:
                     break
             time.sleep(0.2)  # Yield lock for enqueue thread
+
         # Verify all tasks have been completed unless a worker threw an exception
-        while not self._file_upload_queue.empty() and self._exception_queue.empty():
-            time.sleep(0.2)
+        all_ranks_upload_done_tensor = device.tensor_to_device(
+            torch.tensor([int(not self._file_upload_queue.empty() and self._exception_queue.empty())],
+                         dtype=torch.uint8))
+        dist.all_reduce(all_ranks_upload_done_tensor, reduce_operation='MAX')
+        while all_ranks_upload_done_tensor.item() == 1:
+            time.sleep(2)
+            all_ranks_upload_done_tensor = device.tensor_to_device(
+                torch.tensor([int(not self._file_upload_queue.empty() and self._exception_queue.empty())],
+                             dtype=torch.uint8))
+            dist.all_reduce(all_ranks_upload_done_tensor, reduce_operation='MAX')
```

Comment:
for readability maybe alias the boolean to a readable variable name
```suggestion
        uploads_still_not_done  = (all_ranks_upload_done_tensor.item() == 1)
        while uploads_still_not_done:
            time.sleep(2)
            all_ranks_upload_done_tensor = device.tensor_to_device(
                torch.tensor([int(not self._file_upload_queue.empty() and self._exception_queue.empty())],
                             dtype=torch.uint8))
            dist.all_reduce(all_ranks_upload_done_tensor, reduce_operation='MAX')
            uploads_still_not_done  = (all_ranks_upload_done_tensor.item() == 1)
```